frontend/src/modules/events/components/EquipmentSelect.tsx

```diff
@@ -0,0 +1,125 @@
+import { Field, FieldProps } from 'formik';
+import { EquipmentDTO, EventCreateDTO } from 'generated-api';
+import { useContext, useEffect, useRef } from 'react';
+import { useMutation } from 'react-query';
+import { MultiValue } from 'react-select';
+import { SelectAsync } from '../../../shared/components/form/SelectAsync';
+import { filterOptions } from '../../../shared/helpers/filter-options';
+import { isOption } from '../../../shared/helpers/is-option';
+import { ApiContext } from '../../../shared/providers/ApiProvider';
+import { Option } from '../../../shared/types';
+import { formLabelProps } from '../styles';
+
+interface EquipmentSelectProps {
+  defaultValue?: EquipmentDTO[];
+}
+
+export const EquipmentSelect = ({
+  defaultValue: equipment,
+}: EquipmentSelectProps) => {
+  const api = useContext(ApiContext);
+  const equipmentOptions = useRef<Option[]>([]);
+
+  useEffect(() => {
+    fetchEquipment.mutate();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const fetchEquipment = useMutation(() => api.getAllEquipments(), {
+    onSuccess: (data) => {
+      equipmentOptions.current = data.data.map((d) => {
+        return {
+          value: d.id,
+          label: d.name,
+        } as Option;
+      });
+    },
+  });
+
+  const getDefaultValues = (): Option[] => {
+    if (equipment) {
+      const equipmentOptions = equipment.map((eq) => {
+        const option = {
+          value: eq.id,
+          label: eq.name,
+        } as Option;
+
+        return option;
+      });
+
+      return equipmentOptions;
+    }
+
+    return [];
+  };
+
+  /**
+   * Options that allow for async/promise fetching
+   * (but currently, just a simple filtering because we fetch our data on first render)
+   */
+  const promiseOptions = (inputValue: string) =>
+    new Promise<Option[]>((resolve) => {
+      resolve(filterOptions(inputValue, equipmentOptions.current));
+    });
+
+  const handleChange = (
+    newValue: MultiValue<string | Option>,
+    fieldProps: FieldProps
+  ) => {
+    if (newValue === null || undefined) {
+      // replace `null` and `undefined` with empty string for Yup validation
+      fieldProps.form.setFieldValue(fieldProps.field.name, '');
```

Comment:
can we set the initial values for the fields to empty strings instead? so that we don't need this conversion?

Reply:
Empty arrays are indeed set for the formik field props but `React-Select`'s `clear` event resets the values to be of type `null`, causing error tooltips that the value should not be `null` or `undefined`. This is due to the fact that our Formik initial value only accepts empty string or array as "reset" values. 

So, we need to convert React-Select reset values as empty string.